### PR TITLE
MAINT: Bump cyipopt to support new package name and v1.0 API; bump SymPy and SymEngine

### PR DESCRIPTION
**Pull request by bocklund**
Checklist
* [x] The documentation examples have been regenerated if the Jupyter notebooks in the `examples/` have changed. To regenerate the documentation examples, run `jupyter nbconvert --to rst --output-dir=docs/examples examples/*.ipynb` from the top level directory)
* [x] If any dependencies have changed, the changes are reflected in the
  * [x] `setup.py`
  * [x] `environment-dev.yml`

This PR:

- Bumps versions on cyipopt, symengine, python-symengine and sympy
- Updates API of cyipopt for >1.0
- Adds `pip check` to the Test GitHub Action to try to protect against inconsistent dependencies between `setup.py` and those installed by `environment-dev.yml`

### CODE REVIEW

**Comment by bocklund** [comment]
We'll wait to merge this until cyipopt 1.0 is released to pypi and conda-forge. This should allow us to drop the pycalphad channel as cyipopt 0.3 and later should support Windows builds on conda-forge. 

- [ ] drop the pycalphad channel from `environment-dev.yml`
- [ ] drop the pycalphad channel from the installation instruction docs

**Comment by bocklund** [comment]
Once https://github.com/pypa/pypi-support/issues/1008 is closed, cyipopt 1.0 should be released to PyPI shortly after. The 1.0 release steps are being tracked here: https://github.com/mechmotum/cyipopt/issues/94

File: pycalphad/core/solver.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
-import ipopt
-ipopt.setLoggingLevel(50)
+import cyipopt
+cyipopt.setLoggingLevel(50)
```
Comment (bocklund) [inline pycalphad/core/solver.py:2]: This should be `set_logging_level`

Comment (bocklund) [inline pycalphad/core/solver.py:2]:
```suggestion
cyipopt.set_logging_level(50)
```

**Review by richardotis — APPROVED** [review]

**Pull request merged**